gengokit/httptransport/httptransport.go

```diff
@@ -260,9 +260,19 @@ err = json.Unmarshal([]byte({{.LocalName}}Str), {{.LocalName}})`
 		// All repeated args of any type are represented as slices, and bare
 		// assignments to a slice accept a slice as the rvalue. As a result,
 		// LocalName will be declared as a slice, and json.Unmarshal handles
-		// everything else for us.
+		// everything else for us. Addititionally, if a type is a Base type and
+		// is repeated, we first attempt to unmarshal the string we're
+		// provided, and if that fails, we try to unmarshal the string
+		// surrounded by square brackets. If THAT fails, then the string does
+		// not represent a valid JSON string and an error is returned.
 		repeatedUnmarshalTmpl := `
 var {{.LocalName}} {{.GoType}}
+{{- if and (and .IsBaseType .Repeated) (not (Contains .GoType "[]byte"))}}
+err = json.Unmarshal([]byte({{.LocalName}}Str), &{{.LocalName}})
```

Comment:
Currently, truss-to-truss actually will *not* send JSON, it will strip off the square brackets before sending. I will remove the portion which strips braces before sending, as that will keep things nice and fast, like you say.